src/library/analysisfeature.cpp

```diff
@@ -16,13 +16,33 @@
 #include "sources/soundsourceproxy.h"
 #include "util/dnd.h"
 #include "util/debug.h"
+#include "util/logger.h"
 
 const QString AnalysisFeature::m_sAnalysisViewName = QString("Analysis");
 
 namespace {
 
-// Utilize all available cores for batch analysis of tracks
-const int kNumberOfAnalyzerThreads = math_max(1, QThread::idealThreadCount());
+const mixxx::Logger kLogger("AnalysisFeature");
+
+int numberOfAnalyzerThreads() {
+    const int idealThreadCount = QThread::idealThreadCount();
+    DEBUG_ASSERT(idealThreadCount >= 1);
+    // Don't use all available cores on systems with more than
+    // 4 cores to keep the UI and the whole system responsive!
+    if (idealThreadCount <= 4) {
+        // Utilize all available CPU cores
+        return idealThreadCount;
+    } else if (idealThreadCount <= 6) {
+        // Reserve 1 core as headroom
+        return idealThreadCount - 1;
+    } else if (idealThreadCount <= 10) {
+        // Reserve 2 cores as headroom
+        return idealThreadCount - 2;
+    } else {
+        // Never utilize more than 8 cores
```

Comment:
Why do you want to limit it?  
9th gen intel Core i9 has 8 cores with hyperthreading (16 threads) and processors just keep adding more cores ( as one example, my 6th gen i7 "only" has 4 cores/8 threads and the 8th gen has 6 cores/12 threads) . You can add an "idealThreadCount - 3;" in the else.

Reply:
The single UI thread becomes the bottleneck when adding more and more worker threads. Exceeding a certain number of cores stalls the ui as reports indicate. With 4 physical(!) cores the response is acceptable. This is why I decided to use 8 as the limit. 16 already seem to be too much. It is still an 8x speedup compared to 2.2.

Reply:
> Exceeding a certain number of cores stalls the ui as reports indicate.

Where are those reports? I don't recall anyone reporting this. I can't verify that there is any problem that this PR is fixing. I only have computers with 4 or 2 cores.

Reply:
FWIW, when I am batch analyzing, I don't really care if the UI is perfectly responsive as long as it doesn't completely freeze. I care more than the analysis finishes sooner.

Reply:
I do not understand any rationale for these arbitrary limits. You may be right that the UI thread becomes a bottleneck with increasing numbers of worker threads, but no one has actually experimented with this. AFAIK @JosepMaJAZ is the only contributor with a CPU that has >4 cores.